lib/BBTools/utils/BBMapRunner.py

```diff
@@ -196,24 +201,78 @@ def _process_app_params_to_cli(self, io_params, app_params, output_dir, run_log,
                 unmapped_reads_out_file = os.path.join(output_dir, str(assembly_i)+'-'+io_params['out_obj_name']+'-'+'UNMAPPED'+'.FASTQ')
                 options.append('outu={}'.format(unmapped_reads_out_file))
 
-        # hard-code max threads for now
-        options.append('t={}'.format('4'))
+            # add user-config params
+            if app_params.get('input_parameter_suite'):
+                options.append(app_params['input_parameter_suite'])
+
+            if app_params.get('kmer_len') and int(app_params['kmer_len']) != 0:
+                if not app_params.get('input_parameter_suite') \
+                   or 'k=' not in app_params.get('input_parameter_suite'):
+                    options.append("k={}".format(str(app_params['kmer_len'])))
```

Comment:
Ever use f-strings?
This can be written as:
`options.append(f"k={app_params['kmer_len']}")`

Reply:
That's also Python 3 specific. Which this module should be, and if not, should be updated to be.

Reply:
I started the process of updating to the kbase/sdkbase2:python base image, and it started failing.  the first errors, like adding wget and gcc, I could handle, but now there are syntax issues with how to handle imports, etc. that I could probably figure out, but it's likely to turn into serious yak shaving and frankly porting python2 to python3 is not at all my skill set.  Let's absolutely do this in the future, but I'm worried about royally f'ing it up.

Reply:
Yeah, let's get this managed, then we can come back and update to Python 3 later.